docs/project/breaking-change-process.md

```diff
@@ -0,0 +1,44 @@
+
+This process does not result in approval or denial of a breaking change request. It results in feedback, which might be varied or uniform. In the best case, the feedback is a call to action to you, that provides a direction on how you should proceed. The people that are responsible for reviewing your PR are expected to use the information collected by this process as part of their determination of whether your change is appropriate to merge.
+
+You should start this process as soon as possible, preferably before any code is written, in the design phase. The breaking change issue should be paired with a design document that helps reviewers understand the broader scope, goals and constraints of your change. If you don't yet have a design document, you are encouraged to first write such a document, to enable people giving feedback to develop a deeper level of understanding on your approach. If you do not do this, you might stall the process.
+
+There are people on the team that can help you work through a breaking change, if you want to engage with them before starting this more formal process. They are part of the [dotnet/compat team](https://github.com/orgs/dotnet/teams/compat) . Feel free to @mention them on issues to start a dialogue.
+
+## Process
+
+1. Create an issue that describes the breaking change
+   * Mark the issue with the [breaking-change](https://github.com/dotnet/runtime/labels/breaking-change) label
+   * Goals and motivation for the change
+   * Pre-change behavior
+   * Post-change behavior
+   * Errors or other behavior you can expect when running old code that breaks
+   * Workarounds and mitigations, including [AppContext switches](https://docs.microsoft.com/dotnet/api/system.appcontext)
+   * Link to the issue for the feature or bug fix that the breaking change is associated with.
+   * Reference this issue from associated PRs.
+2. Share your issue with whomever you see as stakeholders
+   * Please @mention [dotnet/compat team](https://github.com/orgs/dotnet/teams/compat) team
+   * Engage with people commenting on the issue, with the goal of deriving the most feedback on your proposed breaking change
+   * This may involve a lot of explanation on your part. A well-written design doc can ease the burden here.
+3. Mark associated PRs with the [breaking-change](https://github.com/dotnet/runtime/labels/breaking-change) label, and link to your breaking change issue.
+4. Once the PR is merged, create a [docs issue](https://github.com/dotnet/docs/issues/new?template=dotnet-breaking-change.md).
+   * Clarify which .NET preview the break will be first released in.
```

Comment:
Should we add an announcement in dotnet/announcement too? Presumably this makes sense for some, but not all. How do we decide?

Reply:
I think this would be a rare case.